### PR TITLE
Update export version to handle imports of files without fenced ranges

### LINKED ISSUE #3849: Investigate what happens when importing a table from 2.1 to 3.1
The changes in #3761 changed how files were serialized in the metadata table.  If a table is exported from 2.1 it may have the old file serialization format.  If that table is imported into a 3.1 instance, what happens when it processes the old file serialization format?

### PR DESCRIPTION
**Pull request by EdColeman**
Exports created with versions before 3.1 will not contain range fenced information.   This change allows imports of exported with Accumulo versions prior to 3.1 to be imported into 3.1+.  
  - updated export version to 2 as an optimization so files exported with 3.1+ can be assumed to include fenced ranges.
  - moved the need conversion check to StoredTabletFile for reuse. (also moves tests)

Also verified using uno with exports created with versions 2.1.3 and 3.1 and imported into 3.1 (with these changes)

Closes #3849

### CODE REVIEW

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java [hunk 1]
```diff
@@ -131,8 +132,18 @@ public Repo<Manager> call(long tid, Manager manager) throws Exception {
             Text cq;
 
             if (key.getColumnFamily().equals(DataFileColumnFamily.NAME)) {
-              final StoredTabletFile oldTabletFile = StoredTabletFile.of(key.getColumnQualifier());
-              String oldName = oldTabletFile.getFileName();
+              StoredTabletFile exportedRef;
+              var dataFileCQ = key.getColumnQualifier().toString();
+              if (tableInfo.exportedVersion == null || (tableInfo.exportedVersion < VERSION
+                  && StoredTabletFile.fileNeedsConversion(dataFileCQ))) {
```
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java:138]: This behavior is very tied to a specific version.  I would make that more explicit as follows with a new constant `VERSION_2`.  If in the future the `VERSION` changes this code will still behave the same w.r.t. the specific version where the change in behavior happened.  It also makes it easier to follow the code over time and know what specific version we care about in this if stmt.

```suggestion
              if (tableInfo.exportedVersion == null || tableInfo.exportedVersion < VERSION_2) {
                  checkState(!StoredTabletFile.fileNeedsConversion(dataFileCQ));
```

Comment (EdColeman) [reply server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java]: Added CURR_VERSION and explicit VERSION_2 in 4d136c4c87

**Review by cshannon — COMMENTED** [review]
The changes LGTM but I think it would be good to add a test for import that the conversion works properly. It looks like there is an existing [ImportTableTest](https://github.com/apache/accumulo/blob/68c7e200622ac43498b336a4e7bd4a7bcd1a6888/server/manager/src/test/java/org/apache/accumulo/manager/tableOps/tableImport/ImportTableTest.java) that could maybe be updated:

**Comment by EdColeman** [comment]
The testing difficulty is that the files need to be written in a prior version,  I suppose that the files could be created using 2.1 and added as a resource and preserved in Github.  I'm not aware of other places where we have done that.  Or maybe there is another way?

The changes were testing using uno, but I do agree that is less than satisfying.

**Comment by cshannon** [comment]
Yeah, in this case I think it might be ok to just create small files with 2.1 and commit it as part of the test resources. I have done that before in other projects and I don't think it's a problem here. The only other option is to write code to manually write out the test files each time in the old format which doesn't sound like fun. Since we are just checking if the legacy conversion works seems like just uploading an old file format would be fine. @ctubbsii - Do you have any issue if we uploaded test files created by an older version to check the conversion?

**Comment by keith-turner** [comment]
> Yeah, in this case I think it might be ok to just create small files with 2.1 and commit it as part of the test resources.

We already have some data like that for testing [old rfile formats](https://github.com/apache/accumulo/tree/main/core/src/test/resources/org/apache/accumulo/core/file/rfile).  So there is precedent for that pattern and I think it would be a good thing to do.  Could place a comment in the test showing how the data was produced.

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java [hunk 2]
```diff
@@ -134,7 +134,7 @@ public Repo<Manager> call(long tid, Manager manager) throws Exception {
             if (key.getColumnFamily().equals(DataFileColumnFamily.NAME)) {
               StoredTabletFile exportedRef;
               var dataFileCQ = key.getColumnQualifier().toString();
-              if (tableInfo.exportedVersion == null || (tableInfo.exportedVersion < VERSION
```
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java]: When `tableInfo.exportedVersion < VERSION_2` is true under what circumstance would we expect `StoredTabletFile.fileNeedsConversion(dataFileCQ)` to return false?

**Comment by EdColeman** [comment]
Import test case add in 47f634d3eecae78a.  Includes README in the data dir to create the files with shell commands.

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java (hunk 2 above)
Comment (EdColeman) [reply server/manager/src/main/java/org/apache/accumulo/manager/tableOps/tableImport/PopulateMetadataTable.java]: Removed extra check in 8191010f6

**Comment by EdColeman** [comment]
One quirk of adding the test files where they currently are, the license check shows a warning because it does not know how to check a *.rf file.  (The build completes though)  Unsure if a general exclusion for all rfiles, or something more targeted to the file / directory location would be appropiate.  @ctubbsii do you have a suggestion or preference?  

Another curiosity is that the licence check inserted the ASL into the export distcp.txt file, but it does not seem to matter to the import code.  

The maven warnings:

```
[INFO] --- license:4.3:format (license-headers) @ accumulo-test ---
[INFO] Updating license headers...
[WARNING] Unknown file extension: .../test/src/main/resources/v2_import_test/data/A0000008.rf
[WARNING] Unknown file extension: .../test/src/main/resources/v2_import_test/data/A000000a.rf
[WARNING] Unknown file extension: .../test/src/main/resources/v2_import_test/data/A000000b.rf
[WARNING] Unknown file extension: .../test/src/main/resources/v2_import_test/data/A0000009.rf
[WARNING] Unable to find a comment style definition for some files. You may want to add a custom mapping for the relevant file extensions.
```

**Comment by ctubbsii** [comment]
> Unsure if a general exclusion for all rfiles, or something more targeted to the file / directory location would be appropiate.

Either are fine. I can't imagine we'd ever have a file ending in .rf where we'd want the plugin to add a license header to it, so can probably just exclude them all `**/*.rf` in the parent POM (and remove any more specific .rf exclusions from any child POMs... I know at least one exists in core/pom.xml).

> 
> Another curiosity is that the licence check inserted the ASL into the export distcp.txt file, but it does not seem to matter to the import code.

That one should also be excluded. The format of that file should be one file per line, and the comments it added aren't clearly identifiable as a comment, so we should just avoid it. It also doesn't matter, because there's no copyrightable content in that file anyway. It's just a list of files.

File: test/src/main/java/org/apache/accumulo/test/ImportExportIT.java [hunk 3]
```diff
@@ -380,6 +384,52 @@ private boolean verifyMappingsFile(String destTableId) throws IOException {
...
+
+      int rowCount = 0;
+      try (Scanner s = client.createScanner(table, Authorizations.EMPTY)) {
+        for (Entry<Key,Value> entry : s) {
+          log.trace("data:{}", entry);
+          rowCount++;
+        }
+      }
+      assertEquals(7, rowCount);
+
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ImportExportIT.java:417]: Would be a nice bonus to check the tables splits.

File: test/src/main/resources/v2_import_test/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# Import test data created with Accumulo version 2.1.3
```
Comment (keith-turner) [inline test/src/main/resources/v2_import_test/README.md:22]: This readme is nice.

**Review by keith-turner — APPROVED** [review]

File: test/src/main/java/org/apache/accumulo/test/ImportExportIT.java (hunk 3 above)
Comment (EdColeman) [reply test/src/main/java/org/apache/accumulo/test/ImportExportIT.java]: Added check in 18e8a0ca3e

Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/ImportExportIT.java]: That looks good

**Pull request merged**